packages/dashboard-frontend/src/pages/WorkspaceDetails/OverviewTab/WorkspaceName/index.tsx

```diff
@@ -55,18 +189,143 @@ export class WorkspaceNameFormGroup extends React.PureComponent<Props> {
+        </CheTooltip>
+      );
+    }
+
+    return <div className={styles.helperTextInvalid}>{errorMessage}</div>;
+  }
+
+  private getEditModal(): React.ReactNode {
+    const { isEditModalOpen, editedName, validated } = this.state;
+    const { workspace } = this.props;
+    const isNameChanged = editedName.trim() !== workspace.name;
+    const isDisabled = validated === ValidatedOptions.error || !isNameChanged;
+    const helperTextInvalid = this.getHelperTextInvalid();
+
+    return (
+      <Modal
+        variant={ModalVariant.small}
+        isOpen={isEditModalOpen}
+        title="Edit Workspace Name"
+        onClose={() => this.handleCancelChanges()}
+        actions={[
+          <Button
+            key="save"
+            variant="primary"
+            isDisabled={isDisabled}
+            onClick={() => this.handleSaveChanges()}
+            data-testid="edit-workspace-name-save"
+          >
+            Save
+          </Button>,
+          <Button
+            key="cancel"
+            variant="secondary"
+            onClick={() => this.handleCancelChanges()}
+            data-testid="edit-workspace-name-cancel"
+          >
+            Cancel
+          </Button>,
+        ]}
+      >
+        <FormGroup
+          fieldId="edit-workspace-name"
+          validated={validated}
+          helperTextInvalid={helperTextInvalid}
+          helperTextInvalidIcon={<ExclamationCircleIcon />}
+          className={styles.inputFormGroup}
+        >
+          <TextInput
+            id="edit-workspace-name"
+            value={editedName}
+            onChange={value => this.handleNameChange(value)}
+            validated={validated}
+            maxLength={MAX_LENGTH + 1}
+            data-testid="edit-workspace-name-input"
+            placeholder="Enter workspace name"
+          />
+        </FormGroup>
+      </Modal>
+    );
+  }
+
   public render(): React.ReactNode {
-    const workspaceName = this.buildOpenShiftConsoleLink() || this.props.workspace.name;
+    const { workspace, readonly } = this.props;
+    const workspaceName = this.buildOpenShiftConsoleLink() || workspace.name;
+    const metadataName = workspace.ref.metadata.name;
+
+    if (workspace.name !== metadataName) {
+      return (
+        <>
+          <FormGroup label="Workspace">
+            {readonly ? (
+              <span className={overviewStyles.readonly}>{workspaceName}</span>
+            ) : (
+              <span className={overviewStyles.editable}>
+                {workspaceName}
+                <Button
+                  data-testid="edit-workspace-name-button"
+                  variant="plain"
+                  onClick={() => this.handleEditToggle(true)}
+                  title="Edit Workspace Name"
+                >
+                  <PencilAltIcon />
+                </Button>
+              </span>
+            )}
+            {this.getEditModal()}
+          </FormGroup>
+          <FormGroup label="Metadata Name">
```

Comment:
IMO, exposing k8s DW metadata is confusing, and I would not do that on UD

Reply:
I have fixed it.